build/images/extract.sh

```diff
@@ -43,13 +43,25 @@ function extract-images() {
     customizations="$(get-customizations "$2")"
     [[ -n "$customizations" ]] && flags+=(--set "$customizations")
 
-    {   parallel --nonall --retries 5 --delay 5 helm show chart "${args[@]}" | docker run --rm -i "$YQ_IMAGE" e -N '.annotations."artifacthub.io/images"' -
-        echo '---'
-        parallel --nonall --retries 5 --delay 5 helm template "${args[@]}" --generate-name --dry-run --set "global.chart.name=${2}" --set "global.chart.version=${3}" "${flags[@]}" | tee "${cacheflags[@]}"
-    } | docker run --rm -i "$YQ_IMAGE" e -N '.. | .image? | select(.)' - \
-      | sort -u | sed -e '/^image: null$/d' -e '/^type: string$/d' \
-      | tee >(cat -n 1>&2) >(cat 2>&1 | xargs -n 1 ./inspect.sh 2> /dev/null | cut -f 1 | sed -e "s|^|$(basename $manifest | cut -d. -f 1),$1/$2:$VER,|g" >> $chartmap)
+    # Try to enumerate images via annotations and full manifest rendering
+	
+    {
+
+    images="$( bash <<EOF
+parallel --nonall --retries 5 --delay 5 helm show chart "${args[@]}" | docker run --rm -i "$YQ_IMAGE" e -N '.annotations."artifacthub.io/images"' -
+echo "---"
+parallel --nonall --retries 5 --delay 5 helm template "${args[@]}" --generate-name --dry-run --set "global.chart.name=${2}" --set "global.chart.version=${3}" "${flags[@]}" | tee "${cacheflags[@]}"
+EOF
+)"
+    images="$(printf "%s" "$images" | docker run --rm -i "$YQ_IMAGE" e -N '.. | .image? | .. style="tagged" | select(.)' - | egrep -v 'null|\!\!' | sort -u | xargs || true)"
+    for image in $images
+    do
+	 printf "%s\n" "$image" 
+	./inspect.sh "$image" 2> /dev/null \
+		| cut -f 1  | sed -e "s|^|$(basename $manifest | cut -d. -f 1),$1/$2:$VER,|g" >> $chartmap
+    done 
     
+    } | tee >(cat -n 1>&2)
 }
```

Comment:
Grepping for null will most likely work, but it does not seem very reliable to me. I did a simple test, just adjusting `yq` expression to filter out `image:` fields coming from CRD's. I think that minimal change would suffice to fix issue with kyverno, without ovecomplicating this block (which already looks overcomplicated to me).

Original `yq` expression, which demonstrates the issue:

        $ helm template csm-algol60/cray-kyverno --version 1.2.0 --generate-name --dry-run --set global.chart.name=cray-kyverno --set global.chart.version=1.2.0 | yq4 e -N '.. | .image? | select(.)' -
        description: 'Image is the image name consisting of the registry address, repository, image, and tag. Wildcards (''*'' and ''?'') are allowed. See: https://kubernetes.io/docs/concepts/containers/images.'
        type: string
        image: null
        description: 'Image is the image name consisting of the registry address, repository, image, and tag. Wildcards (''*'' and ''?'') are allowed. See: https://kubernetes.io/docs/concepts/containers/images.'
        type: string
        image: null
        artifactory.algol60.net/csm-docker/stable/ghcr.io/kyverno/kyvernopre:v1.6.2
        artifactory.algol60.net/csm-docker/stable/ghcr.io/kyverno/kyverno:v1.6.2
        artifactory.algol60.net/csm-docker/stable/docker.io/library/busybox:1.28.0-glibc
        artifactory.algol60.net/csm-docker/stable/docker.io/library/busybox:1.28.0-glibc

Slightly modified `yq` expression, which filters out CRDs:

        $ helm template csm-algol60/cray-kyverno --version 1.2.0 --generate-name --dry-run --set global.chart.name=cray-kyverno --set global.chart.version=1.2.0 | yq4 e -N 'select(.kind != "CustomResourceDefinition") | .. | .image? | select(.)' -
        artifactory.algol60.net/csm-docker/stable/ghcr.io/kyverno/kyvernopre:v1.6.2
        artifactory.algol60.net/csm-docker/stable/ghcr.io/kyverno/kyverno:v1.6.2
        artifactory.algol60.net/csm-docker/stable/docker.io/library/busybox:1.28.0-glibc
        artifactory.algol60.net/csm-docker/stable/docker.io/library/busybox:1.28.0-glibc

So, I think the issue can be solved by just adding `select(.kind != "CustomResourceDefinition") | ` into `yq` expression at line 49.

Reply:
@mtupitsyn I think that's a good optimization on face value. I'm somewhat concerned we'll trip over other 'false' data structures that trigger on image, but none come to mind. I'll pull this in to test and circle back.

Don't disagree re: complexity, but I'm not looking to do wholesale refactoring of this logic in this PR.

Reply:
Tested `yq` query as part of original chain (e.g. `make` and `extract.sh`) - it looks like we need to add `?` to `.kind`. I.e. overall `yq` expression in line 49 should be:

        } | docker run --rm -i "$YQ_IMAGE" e -N 'select(.kind? != "CustomResourceDefinition") | .. | .image? | select(.)' - \